### PR TITLE
Plumb node funding from genesis

### PR DESCRIPTION
**Pull request by danpaul000**
Existing scripts were not correctly propagating genesis node funding to fullnode.sh.  This should fix it!

Also, add explicit airdrop enable/disable option, rather than implying it by the presence of genesis node funding.  Explicitly setting node stake, node funds at genesis and airdrops on/off are now supported from top level CI scripts.

### CODE REVIEW

**Comment by danpaul000** [comment]
TdS pipeline default settings are updated to reflect this change, though I'm not sure what default value to use for LAMPORTS_INTERNAL_NODES.  Currently I have it set to:
```
  DISABLE_AIRDROPS: "true"
  STAKE_LAMPORTS_INTERNAL_NODES: "8589934592"
  LAMPORTS_INTERNAL_NODES: "17000000000"  
```

File: net/net.sh [hunk 1]
```diff
@@ -137,13 +143,19 @@ while [[ -n $1 ]]; do
     elif [[ $1 = --deploy-update ]]; then
       updatePlatforms="$updatePlatforms $2"
       shift 2
-    elif [[ $1 = --stake-internal-nodes ]]; then
+    elif [[ $1 = --stake-lamports-internal-nodes ]]; then
       stakeNodesInGenesisBlock="$2"
```
Comment (mvines) [inline net/net.sh]: Rename `stakeNodesInGenesisBlock` to `internalNodeStakeLamports`?   This no longer has anything to do with the genesis block

File: net/net.sh [hunk 2]
```diff
@@ -137,13 +143,19 @@ while [[ -n $1 ]]; do
...
       shift 2
+    elif [[ $1 = --lamports-internal-nodes ]]; then
+      fundNodesInGenesisBlock="$2"
```
Comment (mvines) [inline net/net.sh]: Rename `fundNodesInGenesisBlock` to `internalNodeLamports`?

File: net/remote/remote-node.sh [hunk 3]
```diff
@@ -29,7 +31,11 @@ export RUST_LOG
 # trouble
 #
 # Ref: https://github.com/solana-labs/solana/issues/3798
-stake=424243
+stake=${stakeNodesInGenesisBlock:=424243}
+
+# TODO: There is a default stake set here, but the existing static/default for node_lamports/fundNodes is in fullnode.sh.
+# TODO: Should there be a default value this deep if we are no longer assuming airdrops one way or the other?
+#fundNodesInGenesisBlock=${fundNodesInGenesisBlock:=0}
```
Comment (mvines) [inline net/remote/remote-node.sh]: Na, I'd say remove

File: net/remote/remote-node.sh [hunk 4]
```diff
@@ -154,20 +160,21 @@ local|tar)
     if [[ $skipSetup != true ]]; then
       args=(
         --bootstrap-leader-stake-lamports "$stake"
+        --bootstrap-leader-lamports "$fundNodesInGenesisBlock"
```
Comment (mvines) [inline net/remote/remote-node.sh]: This should be wrapped in a `if [[ -n $fundNodesInGenesisBlock ]]; then` block right?

File: net/remote/remote-node.sh [hunk 5]
```diff
@@ -202,13 +209,16 @@ local|tar)
     else
       args+=(--stake "$stake")
       args+=(--enable-rpc-exit)
+      if [[ -n $fundNodesInGenesisBlock ]] ; then
```
Comment (mvines) [inline net/remote/remote-node.sh]: nit: remove space before `;`

**Comment by danpaul000** [comment]
Validators failed to start with
```
current account balance is 
8589934592 lamports
4BvDbKdXGCqJb3dwTNzXDGpnQqnyPFJoNGDFjTVZ7V1N8GVvdau7MAoSvrMKieikLGcM8JGSpbZ3VmZnwG71gwSg
Error: TransactionError(InstructionError(0, CustomError(1)))
```

I set the buildkite vars to have 17B lamports and 8B stake, but looks like this node got 8B lamports.  I must have crossed the streams somewhere.  Will debug in the AM.

File: net/remote/remote-node.sh (hunk 4 above)
Comment (danpaul000) [reply net/remote/remote-node.sh]: yes

File: net/remote/remote-node.sh [hunk 6]
```diff
@@ -109,13 +109,13 @@ local|tar)
...
-    if [[ -n $internalNodesStakeLamports ]]; then
+    rm -rf ./solana-node-balances
+    mkdir ./solana-node-balances
+    if [[ -n $internalNodesLamports ]]; then
       for i in $(seq 0 "$numNodes"); do
         solana-keygen new -o ./solana-node-keys/"$i"
         pubkey="$(solana-keygen pubkey ./solana-node-keys/"$i")"
-        echo "${pubkey}: $internalNodesStakeLamports" >> ./solana-node-stakes/fullnode-stakes.yml
+        echo "${pubkey}: $internalNodesLamports" >> ./solana-node-balances/fullnode-balances.yml
       done
```
Comment (danpaul000) [inline net/remote/remote-node.sh]: Bit of a bug here.

File: net/remote/remote-node.sh [hunk 7]
```diff
@@ -184,6 +184,7 @@ local|tar)
   validator|blockstreamer)
     net/scripts/rsync-retry.sh -vPrc "$entrypointIp":~/.cargo/bin/ ~/.cargo/bin/
     rm -f ~/solana/fullnode-identity.json
+    # TODO: Is this supposed to be Lamports or StakeLamports?
```
Comment (danpaul000) [inline net/remote/remote-node.sh]: @mvines  ^

Comment (mvines) [reply net/remote/remote-node.sh]: Lamports, not StakeLamports

**Pull request merged**